### PR TITLE
Add Postgres integration repository

### PR DESCRIPTION
**Pull request by jgayfer**
This is still pretty rough, but establishes a basic pattern of a Postgres repository, along with migrations.

Wanted to get it up for a look since there's a lot of "establishing" going on.

Some things I'd like to clean up before merge:
- Proper database config
- Better test database setup / cleaning between runs

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 [workspace]
 members = [
     "core",
+    "hold",
```
Comment (derrickp) [inline Cargo.toml:4]: out of curiosity, why "hold"?

File: hold/migrations/20220713044015_create_integrations_table.sql [hunk 2]
```diff
@@ -0,0 +1,6 @@
+-- Add migration script here
+CREATE TABLE integrations(
```
Comment (derrickp) [inline hold/migrations/20220713044015_create_integrations_table.sql:2]: More a question of the file name than the contents, but do you have to have a specific format for the filenames? Or is it as long as they can be sorted? Does sqlx store the migrations it has run in the database?

File: hold/Cargo.toml [hunk 3]
```diff
@@ -6,3 +6,8 @@ edition = "2021"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
+async-trait = "~0.1"
+lfg-core = { path = "../core" }
+lfg-support = { path = "../support" }
+sqlx = { version = "0.6", features = [ "runtime-tokio-native-tls" , "postgres" ] }
+tokio = { version = "1.0", features = ["full"] }
```
Comment (derrickp) [inline hold/Cargo.toml:13]: I cannot remember, does this mean that anything that is `1.0.X` is ok? Or would `1.19.2` also work?

File: hold/tests/integration_repo.rs [hunk 4]
```diff
@@ -0,0 +1,51 @@
+use lfg_core::repo::IntegrationRepo;
+use lfg_hold::repo::PsqlIntegrationRepo;
+use lfg_support::fixtures::TestNewIntegration;
+use sqlx::postgres::PgPoolOptions;
+
+#[tokio::test]
```
Comment (derrickp) [inline hold/tests/integration_repo.rs:6]: Is there a way to mark these as only runnable locally? Or I wonder if we could run something else (like sqlite) in CI. Not sure how much postgres specific stuff we'll use.

File: hold/tests/integration_repo.rs [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+async fn find_empty() -> Result<(), sqlx::Error> {
+    let pool = PgPoolOptions::new()
+        .connect("postgres://jgayfer:@localhost/lfg")
```
Comment (derrickp) [inline hold/tests/integration_repo.rs:9]: we could add a DB url I guess.

**Comment by derrickp** [comment]
I assume if most things used a transaction we could just not commit the transaction and roll it back and all of the tests should work? I'm not actually sure there. I think I'd rather not just truncate the tables in between, but it's something we could do if we needed to.

File: Cargo.toml (hunk 1 above)
Comment (jgayfer) [reply Cargo.toml:4]: It's copied from the Apollo project, where the idea was it was more of "the hold", like on a ship, except a rocket ship (Apollo mission).

But I've kind of cheated by stealing it, and would like a clever name!

File: hold/migrations/20220713044015_create_integrations_table.sql (hunk 2 above)
Comment (jgayfer) [reply hold/migrations/20220713044015_create_integrations_table.sql:2]: I imagine the timestamp it was matters (sqlx generates it), and then yes, sqlx stores it in the database to keep track of what has been run.

I'll add some notes on how to use it before I merge.

```
jgayfer@/tmp:lfg> \d _sqlx_migrations;
+----------------+--------------------------+-------------------------+
| Column         | Type                     | Modifiers               |
|----------------+--------------------------+-------------------------|
| version        | bigint                   |  not null               |
| description    | text                     |  not null               |
| installed_on   | timestamp with time zone |  not null default now() |
| success        | boolean                  |  not null               |
| checksum       | bytea                    |  not null               |
| execution_time | bigint                   |  not null               |
+----------------+--------------------------+-------------------------+
Indexes:
    "_sqlx_migrations_pkey" PRIMARY KEY, btree (version)
```

File: hold/Cargo.toml (hunk 3 above)
Comment (jgayfer) [reply hold/Cargo.toml:13]: It looks like `1.19.2` would work; I believe anything up to `2.0` would work.

From the cargo docs:
```
1.2.3  :=  >=1.2.3, <2.0.0
1.2    :=  >=1.2.0, <2.0.0
1      :=  >=1.0.0, <2.0.0
0.2.3  :=  >=0.2.3, <0.3.0
0.2    :=  >=0.2.0, <0.3.0
0.0.3  :=  >=0.0.3, <0.0.4
0.0    :=  >=0.0.0, <0.1.0
0      :=  >=0.0.0, <1.0.0

```

**Comment by jgayfer** [comment]
@derrickp I might try what zero2prod does, where he actually just spins up a new database for each test run.. could fail horribly but going to hear him out and see what happens!

File: hold/tests/integration_repo.rs (hunk 4 above)
Comment (jgayfer) [reply hold/tests/integration_repo.rs:6]: Good question... I could see there being a lot of Postgres tests. Especially if and when end to end request specs are using postgres. Not sure what the best call there is yet.

Comment (derrickp) [reply hold/tests/integration_repo.rs:6]: Yeah, I'm not sure if github actions would let you spin up an environment. Maybe via docker?

**Pull request closed without merging**